test/unit/deferred.js

```diff
@@ -542,6 +543,26 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ typeof Symbol === "function" ? "test" : "skip" ](
+	"jQuery.Deferred.then - IsCallable determination (gh-3596)",
+	function( assert ) {
+
+		assert.expect( 1 );
+
+		var done = assert.async(),
+			defer = jQuery.Deferred();
+
+		function faker() {
+			assert.ok( true, "handler with non-'Function' @@toStringTag gets invoked" );
+		}
+		faker[ typeof Symbol === "function" && Symbol.toStringTag ] = "String";
```

Comment:
Why not wrap the statement with `if`? In the current way we'll be writing to a `"false"` property in non-supporting browsers which may create confusion.

These are tests so even size is not important so I don't see the gains of the current form.